### DISCUSSION TITLE
Possible to add additional response to schema from dependency?

### DISCUSSION THREAD

**Discussion by jeremyschiemann** [fastapi/fastapi#11144] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import fastapi

app = fastapi.FastAPI()

def dependency_403():
    raise fastapi.HTTPException(403, 'Forbidden')

@app.get(
    '/',
    dependencies=[Depends(dependency_403)],
    responses={
        403: {'description': 'Forbidden'},
    },
)
def root() -> str:
    return 'Hello World'

# many more routes, using the dependency
```

### Description

I'm having a dependency that can raise a 403, to represent this in the OpenAPI schema, I have to add an additional response to every endpoint using the dependency manually.

Is there a way to declare dependency that automatically adds an extra response to the endpoints using it?  (Or am I handling that wrong?)

I'd imagine something like this:

```python

# Adds 403 response to every endpoints schema (or router)
app = FastAPI(
  dependencies=[
    Depends(my_dependency, additional_response={403: {'description': 'Forbidden'}}),
  ]
)
```

### Operating System

Linux

### FastAPI Version

0.108.0

### Pydantic Version

2.5.3

### Python Version

3.11

**Comment by Kludex** [comment]
- Duplicated of https://github.com/tiangolo/fastapi/discussions/9124

**Reply by wont-work** [reply]
This is not a duplicate. OP asked if a *dependency* can add extra responses. That discussion is about modifying the route handlers themselves. 

If I have a dependency that does, say, authentication (just an example, I know fastapi has something builtin for that), do you really expect me to copy paste the unauthenticated error response to every single handler that depends on said dependency? How about dependencies taht depend on said authentication dependency? It'd be extremely easy for the OpenAPI schema to go out of sync from the actual logic.

If it's not possible just say so and move on.

**Comment by galah92** [comment]
Was anyone able to find an elegant solution for this?

**Comment by Sagor0078** [comment]
I think FastAPI does not natively support automatically injecting additional OpenAPI responses from dependencies

**Comment by YuriiMotov** [comment]
There is already issue to track this: https://github.com/fastapi/fastapi/issues/501
